### PR TITLE
gradle2nix v2

### PR DESCRIPTION
**Pull request by rgoulter**

### CODE REVIEW

**Comment by rgoulter** [comment]
Not sure why, but `builders.aarch64-darwin` isn't in the gradle2nix flake output, even though the `apps` have `aarch64-darwin`, and there's `builders.x86_64-darwin`, `builders.x86_64-linux`.

**Comment by rgoulter** [comment]
Though, using the gradle2nix's `packages.aarch64-darwin` doesn't help, either.

**Comment by rgoulter** [comment]
The specific error output: (e.g. https://github.com/rgoulter/c-worksheet-instrumentor/actions/runs/9302966681/job/25604253344)

```
copying path '/nix/store/dc477ngpdyz4wqn5yswppahrvr0vcghy-source' from 'https://cache.nixos.org/'...
error:
       … while evaluating the attribute 'packages.aarch64-darwin.c-worksheet-instrumentor'
         at /nix/store/rggwdz1hhylyrn8s210vaxy7llknh8w8-source/flake.nix:50:7:
           49|
           50|       c-worksheet-instrumentor = pkgs.callPackage ./c-worksheet-instrumentor.nix {inherit (gradle2nix) buildGradlePackage;};
             |       ^
           51|     });

       … while calling a functor (an attribute set with a '__functor' attribute)
         at /nix/store/dc477ngpdyz4wqn5yswppahrvr0vcghy-source/lib/customisation.nix:264:13:
          263|     in if missingArgs == {}
          264|        then makeOverridable f allArgs
             |             ^
          265|        # This needs to be an abort so it can't be caught with `builtins.tryEval`,

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: attribute 'aarch64-darwin' missing
       at /nix/store/rggwdz1hhylyrn8s210vaxy7llknh8w8-source/flake.nix:46:20:
           45|       pkgs = nixpkgs.legacyPackages.${system};
           46|       gradle2nix = inputs.gradle2nix.packages.${system}.gradle2nix;
             |                    ^
           47|     in {
```

My initial thought was something about the `builders` flake output; but, switching to the `packages` didn't improve things. (It still builds on x86_64-linux and x86_64-darwin, while failing on aarch64-darwin).

This is confusing:

- aarch64-darwin shows in the output for the gradle2nix flake on the v2 branch.
- There's nothing specific to gradle2nix, flake-utils, or its systems/default that'd exclude aarch64-darwin.

**Comment by rgoulter** [comment]
For the `aarch64-darwin` build, it wasn't enough to refer to `inputs.gradle2nix`; the input had to be renamed to something else: https://github.com/rgoulter/c-worksheet-instrumentor/commit/2acc2938696dd6eb2c67538085058e9c883d9e8c

File: flake.nix [hunk 1]
```diff
@@ -31,16 +35,18 @@
 
     devShells = forAllSystems (system: let
       pkgs = nixpkgs.legacyPackages.${system};
+      gradle2nix = inputs.gradle2nixF.builders.${system};
```
Comment (tadfisher) [inline flake.nix]: Ah, another fix is

```nix
inherit (inputs.gradle2nix.builders.${system}) buildGradlePackage;
```

**Pull request merged**